junit5-extension/src/test/java/io/kroxylicious/testing/kafka/junit5ext/TemplateTest.java

```diff
@@ -48,46 +46,20 @@ static Stream<BrokerCluster> clusterSizes() {
                 brokerCluster(3));
     }
 
-    static Map<Integer, Integer> observedMultipleClusterSizes = new HashMap<>();
-
-    @Nested
-    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
-    public class MultipleClusterSizes {
-        @TestTemplate
-        public void testMultipleClusterSizes(
-                                             @DimensionMethodSource(value = "clusterSizes", clazz = TemplateTest.class) KafkaCluster cluster)
-                throws ExecutionException, InterruptedException {
-            try (var admin = Admin.create(cluster.getKafkaClientConfiguration())) {
-                observedMultipleClusterSizes.compute(admin.describeCluster().nodes().get().size(),
-                        (k, v) -> v == null ? 1 : v + 1);
-            }
-        }
-
-        @AfterAll
-        public void afterAll() {
-            assertEquals(Map.of(1, 1, 3, 1),
-                    observedMultipleClusterSizes);
+    @TestTemplate
+    public void testMultipleClusterSizes(
+                                         @DimensionMethodSource(value = "clusterSizes", clazz = TemplateTest.class) KafkaCluster cluster)
+            throws ExecutionException, InterruptedException {
+        try (var admin = Admin.create(cluster.getKafkaClientConfiguration())) {
+            assertEquals(admin.describeCluster().nodes().get().size(), cluster.getNumOfBrokers());
         }
     }
 
-    static Map<Integer, Integer> observedMultipleClusterSizesWithAdminParameters = new HashMap<>();
-
-    @Nested
```

Comment:
Why are we loosing the `@Nested` annotation here?

Reply:
It is not needed anymore as we can assert in each iteration of the template and not waiting till the end of the loop to assert in the `@AfterAll` section that honestly I don't like very much

Reply:
Ahh Sweet, That's great!